subworkflows/nf-core/bed_scatter_bedtools/main.nf

```diff
@@ -0,0 +1,81 @@
+include { BEDTOOLS_SPLIT     } from '../../../modules/nf-core/bedtools/split/main'
+include { GAWK               } from '../../../modules/nf-core/gawk/main'
+
+workflow BED_SCATTER_BEDTOOLS {
+
+    // IMPORTANT: Add the configuration in nextflow.config file to your modules.config
+
+    take:
+    ch_bed          // channel: [ meta, bed, scatter_count ]
+    ch_fasta_fai    // channel: [ fasta_fai ]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    ch_bed
```

Comment:
Instead of branching, shouldn't we make both inputs optional?
That way we split a bed file either way, but we generate one first if a `fai` is given.

Reply:
agree here.

ch_input = bed.mix(GAWK(fasta).out.output)

Reply:
Shouldn't that generation from fasta done at the pipeline level, outside of this subworkflow?

Reply:
That would be preferable yes, but then the entire submodule is kind of moot, as it's just a bedtools split 😅

Reply:
In term of processes yes I agree. But for me the most important part is the channel operations that's the magic we want to share in between pipelines.

Reply:
Works for me! In that case, I'd toss the fai dependency and just go with the bed stuff 👍🏻

Reply:
Allright! I'll change it when I got some time :)

Reply:
There really isn't that much code left indeed, even operation wise it's pretty simple.